src/web-auth/index.js

```diff
@@ -1146,10 +1146,18 @@ WebAuth.prototype.passwordlessVerify = function (options, cb) {
  * @param {Function} [options.templates.error] template function returning a custom error message when the challenge could not be fetched, receives the error as first argument
  * @param {String} [options.lang=en] the ISO code of the language for the captcha provider
  * @param {Function} [callback] An optional completion callback
+ * @param {Object} [arkoseConfig] An optional configuration object for arkose captcha
+ * @param {String} [arkoseConfig.onCompleted] function fired after Arkose challenge is completed
+ * @param {String} [arkoseConfig.onError] nction fired after Arkose encounters an error with the error passed as first argument
  * @memberof WebAuth.prototype
  */
-WebAuth.prototype.renderCaptcha = function (element, options, callback) {
-  return captcha.render(this.client, element, options, callback);
+WebAuth.prototype.renderCaptcha = function (
```

Comment:
I see we are adding an explicit `arkoseConfig`, what happens if we need additional configuration for future supported captcha providers?

Additionally, I see the arkoseCOnfig takes an `onCompleted`, how is this diffferend from the already existing `callback` parameter ?

Reply:
I also see most existing captcha providers have an onError hook, which isnt used to pass through to the caller. 

Not saying we should implement that, but I do wonder if it would make sense to get rid of the explicit arkose mention here, and maybe just use the following signature:

```js
function (element, options, callback, errorCallback)
```

Reply:
Additionally, the question could be why we need to be made aware of any errors in the case or arkose and not in the case of any other captcha provider.

Reply:
> I see we are adding an explicit arkoseConfig, what happens if we need additional configuration for future supported captcha providers?

You make a good point. This may not be best for extensibility. I'm open to other suggestions on how to approach this.

> Additionally, I see the arkoseCOnfig takes an onCompleted, how is this diffferend from the already existing callback parameter ?

The callback parameter is something we call after the captcha has finished loading into the UI. The `onCompleted` inside arkoseConfig is a custom callback that the user can provide and Arkose will call it when a captcha has been solved.

> Not saying we should implement that, but I do wonder if it would make sense to get rid of the explicit arkose mention here, and maybe just use the following signature:

We'll need the other callbacks for Arkose, especially `onCompleted`. The user will likely want to fire a submit through their `onCompleted` callback.

> Additionally, the question could be why we need to be made aware of any errors in the case or arkose and not in the case of any other captcha provider.
It's just an option for the user to know the error for Arkose. Since we plan to give the user callbacks for `onComplete` and `onReady`, we may as well give a `onError` callback option. Another possibility is that the user may want to add some logic in the case of Arkose errors. It's important to note that Arkose needs to be more configurable since the user will fire the captcha through an API call unlike other captchas where there's a checkbox or button to click. As for other captchas, it's just how they've been and I don't plan to changing those as part of this PR.

Reply:
> As for other captchas, it's just how they've been and I don't plan to changing those as part of this PR.

Yes, sorry that wasnt what I mean. Just saying that theoretically they could use that as well. They don't, and they shouldnt in this PR. It just makes me wonder whether it needs to be named specifically for arkose, or whether we can have a more generic name?

I think it would benefit to use something more generic, but I understand there may be provider specific hooks as well.

Regarding onError, theoretically, in the future, one could wire up this as well: https://github.com/auth0/auth0.js/blob/master/src/web-auth/captcha.js#L198-L200, which could use the same onError, as I think this serves the same purpose as the arkose one, right?

> The callback parameter is something we call after the captcha has finished loading into the UI. The onCompleted inside arkoseConfig is a custom callback that the user can provide and Arkose will call it when a captcha has been solved.

Yeah, I noticed the callback is not called when the captcha provider calls onComplete. So I think the original jsdocs (` * @param {Function} [callback] An optional completion callback`) is confusing with the newly added `onCompleted`. How does it differ without looking at the source code? 

What if we :

- Change the JSDoc of the callback to indicate its not about completed, but finished loading?
- add the config like it is, but drop arkose from it.
- Add a jsdoc to indicate it is currently only used for arkose, or wire it to the corresponding methods.
- Additionally, we could wire them to the others as well. But I understand this is not the scope of this PR so I leave that to you to decide.
- If we ever need provider specific hooks, we can add provider specific keys to the config object: `{ onCompleted: ..., arkose: { arkoseSpecificHook: ... }}`

Reply:
In generally I think we shouldnt overthink making it generic, but also want to avoid making it explicit like this as it complicates things going forward. Also want to keep in mind that we can make future changes in a non-breaking way where possible.

Having the parameter named arkoseConfig is something we can actualy keep for now if we want. It's none breaking to rename that. But it kinda reds a bit weird looking at this function from a generic point of view.

Reply:
Something else we can do ... What about adding `arkoseConfig` to the existing `options`? We are then adding a provider specific config object in a generic options object. Sounds like a good middleground?

Reply:
updated following discussion